src/process_utility.c

```diff
@@ -282,9 +283,20 @@ check_altertable_add_column_for_compressed(Hypertable *ht, ColumnDef *col)
 					has_notnull = true;
 					continue;
 				case CONSTR_DEFAULT:
-					/* since default expressions might trigger a table rewrite we
-					 * only allow Const here for now */
-					if (!IsA(constraint->raw_expr, A_Const))
+					/*
+					 * Since default expressions might trigger a table rewrite we
+					 * only allow Const here for now.
+					 */
+					if (IsA(constraint->raw_expr, TypeCast))
+					{
+					   /*
+						* Ignore error for boolean column as values like
+						* 'True' or 'False' are treated as TypeCast.
+						*/
+						char *name = strVal(llast(((TypeCast *) constraint->raw_expr)->typeName->names));
+						is_bool = strcasestr(name, "bool") ? true : false;
```

Comment:
This feels too specific -- e.g. can I spell it as 'boolean'? What about casts for other types? etc

Maybe instead we could call `eval_const_expressions` before performing the old check?

Reply:
This is actually added with that exact spelling by parser. If you look at the tests you can see that the column definition actually uses a different spelling for the column type. This is special casing because the parser treats `'f'` different from `false` or `true` so if you do `ADD COLUMN foo DEFAULT false` it will fail the Const check but `ADD COLUMN foo DEFAULT 'f'` it will be represented as Const. This is specific to bool handling and all the other datatypes should be covered by the Const-check.

Reply:
Well it doesn't matter what is given in the query, because this value is always bool. TypeName->names is a list of schema and type which is 'pg_catalog' and 'bool' respectively.
Also, so far we only know about this case where we want to allow.

Reply:
To make it bulletproof we should also check that the typecast-arg is a Const, we dont want function calls or anything else here.

Reply:
While using `eval_const_expressions` would be the most generic solution it cannot be used here because it only works on planner nodes and the nodes we are dealing with here are parser nodes.